Astro Teams Docs (#2328)
astronomer/docs: astro/user-permissions.md

description: Learn about Astronomer's RBAC system and how to assign roles to users.
---

> To better protect your data pipelines and cloud infrastructure, Astro provides role based access control (RBAC) for Organizations and Workspaces. Each Astro user has a Workspace role in each Workspace they belong to, plus a single Organization role. Organization Admins can create _Astro Teams_ of individual users, and then assign those teams of users the same Workspace role level. Role based access control is not available for Deployments.

Reviewer: ```suggestion
To better protect your data pipelines and cloud infrastructure, Astro provides role based access control (RBAC) for Organizations and Workspaces. Each Astro user has a Workspace role in each Workspace they belong to, plus a single Organization role. Users can also belong to [Teams](add-user.md#make-a-team), which apply the same Workspace role to a group of users. Role based access control is not available for Deployments.
```

- Organization Owner > Organization Billing Admin > Organization Member 
- Workspace Admin > Workspace Editor > Workspace Member

> Additionally, Organization Owners inherit Workspace Admin permissions for all Workspaces in the Organization and only Organization Owners can create Teams.

Reviewer: ```suggestion
Additionally, Organization Owners inherit Workspace Admin permissions for all Workspaces in the Organization.
```
Doesn't seam like the right place to include this, given that it's already in the table below. This sentence has more to do about inheritance than general permissions.

| Remove a user from an Organization                               |                         |                                | ✔️                      |
| Create, update and delete Organization API tokens                                   |                         |                                | ✔️                      |
| Access, regenerate, and delete single sign-on (SSO) bypass links |                         |                                | ✔️                      |

> | Create a Team of users |                         |                                | ✔️                      |

Reviewer 2: It's more then Create... should we specify full CRUD similar to Org API tokens line?

| Create, Update and Delete API Keys                  |                      |                      | ✔️                   |
| Update user roles and permissions                   |                      |                      | ✔️                   |
| Invite users to a Workspace                         |                      |                      | ✔️                   |

> | Assign Teams to Workspaces or Deployments         |                      |                      | ✔️                   |

Reviewer 2: Assign and Remove?

Reviewer 2: To Workspaces only. I don't want to confuse customers thinking they can apply Teams to specific Deployments.

| Create, update and delete Workspace API tokens                         |                      |                      | ✔️                   |

To manage a user's Workspace permissions, see [Manage users](add-user.md#add-a-user-to-a-workspace).

> ## Roles and Teams

Reviewer: ```suggestion
## User roles and Team roles
```

> There are two ways to define user roles for a Workspace:

Reviewer: ```suggestion
There are two ways to define a user's role in a Workspace:
```

> - Define the individual user role when you [add a user](/astro/add-user.md#add-a-user-to-a-workspace) to a Workspace 

Reviewer: ```suggestion
- Define the individual user role when you [add a user](/astro/add-user.md#add-a-user-to-a-workspace) to a Workspace.
```

> - Assign a Workspace role to a [team of users](ADD LINK)

Reviewer: ```suggestion
- Assign a Workspace role to a [Team](/astro/add-user.md#add-a-team-to-a-workspace).
```

> If the user role that you assign at the Team level is different from what you assigned the user individually, then Astronomer recognizes the more privileged role.

Reviewer: ```suggestion
If a user has permissions to a Workspace both as an individual and as a member of a Team, then Astronomer recognizes the more privileged role.
```

> For example, if an Organization Owner first adds a user to a Workspace as an **Organization Member** and a **Workspace Member**. Then, they assign this user to a Team of users with **Workspace Admin** privileges, this user now has **Workspace Admin** privileges for any Workspaces the Team is added to.

Reviewer: ```suggestion
For example, if a user belongs to a Workspace as a **Workspace Member**, but also belongs to a Team in the Workspace with **Workspace Admin** privileges, then the user has **Workspace Admin** privileges in the Workspace.
```
Tried cutting out some of the "adding" details so that we can focus on the privilege relationship here.
